app/controllers/carts_controller.rb

```diff
@@ -135,7 +135,11 @@ def process_cart_update action_type
   end
 
   def increment_cart_quantity
-    stock = @cart.product.stock
+    if @cart.product.nil?
+      flash[:warning] = t "flash.not_found_product"
+      redirect_to root_path and return
```

Comment:
and return này để làm gì thế?

Reply:
sau redirect_to dùng and return để không chạy lệnh phía sau nữa